### PR TITLE
Cleanup CI workflow

### PR DESCRIPTION
**Pull request by quexten**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: Rust
 
 on:
   push:
-    branches: [ main ]
+    branches: [main]
```
Comment (withinfocus) [inline .github/workflows/rust.yml:5]: ⛏️ The brackets are unnecessary but I would surround with double quotes.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -14,62 +14,57 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
```
Comment (withinfocus) [inline .github/workflows/rust.yml:17]: ❌ Needs a version tag after the hash.

**Review by withinfocus — CHANGES_REQUESTED** [review]

File: .github/workflows/rust.yml (hunk 2 above)
Comment (quexten) [reply .github/workflows/rust.yml]: Fixed in [4588d8f](https://github.com/bitwarden/bitwarden-russh/pull/5/commits/4588d8f4afd966fbd7be259cf3c24dd4999c3a97)

File: .github/workflows/rust.yml (hunk 1 above)
Comment (quexten) [reply .github/workflows/rust.yml]: Cleaned up in [4588d8f](https://github.com/bitwarden/bitwarden-russh/pull/5/commits/4588d8f4afd966fbd7be259cf3c24dd4999c3a97)

**Review by withinfocus — APPROVED** [review]

**Pull request merged**